server/api/cvrs.py

```diff
@@ -652,36 +644,31 @@ def parse_ballots_file(
                 row, "Cast Vote Record", row_index + 1, header_indices
             )
             batch_name = column_value(row, "Batch", cvr_number, header_indices)
+
+            # Tabulator CVR is either a 10-digit string or a 16-character hex ID.
+            #
+            # When it's a 10-digit string, the first four digits are a tabulator ID, and the last
+            # six are the ballot number (record ID) within the batch. We use the full value as an
+            # imprinted ID even though it's not imprinted on the ballots.
+            #
+            # When it's a 16-character hex ID, it's actually imprinted on the ballots. We construct
+            # a ballot number (record ID) from the hex ID even though we don't know if it
+            # corresponds to ballot order.
+            #
+            # When the ballots file has a Machine column, we use that as a tabulator ID. Otherwise,
+            # we have to use the Tabulator CVR column for this purpose, and all Tabulator CVR
+            # values have to be 10-digit strings.
+            #
```

Comment:
Nothing new in this comment - just paraphrasing existing comments below and removing references to type-1 and type-2 ballots files